libs/libc/stdlib/lib_exit.c

```diff
@@ -39,9 +39,9 @@ void exit(int status)
 {
   atexit_call_exitfuncs(status);
 
-  /* REVISIT: Need to flush files and streams */
+  /* Call nx_exit() which will do the rest of the clean up and exit */
 
-  _exit(status);
+  nx_exit(status);
```

Comment:
why change _exit to nx_exit?

Reply:
It does the file stream clean up etc, like before.
Calling _exit() does not work it in fact results in system crash.

Reply:
But, the final goal is remove

> Thanks for the comments, I'll try to tidy up the exit() functions a bit better. My idea was to do this temporary patch first and then clean everything up in the next PR, but I can try to do this PR better.
> 
> > But why not enable the compile of lib_exit?
> 
> Well I can of course do that, just need to omit the call to the exit functions. So if this is OK to you I can do it like this:
> 
> ```
> void exit(int status)
> {
> #if 0 /* REVISIT: Enable this when kernel side atexit/on_exit is removed */
>   atexit_call_exitfuncs(status);
> #endif
>   /* Call nx_exit() which will do the rest of the clean up and exit */
> 
>   nx_exit(status);
> }
> ```

How about create a new patch in this PR to remove the exit callback from kernel side? And remove the kernel atexit implementation, and enable libc's version.

Reply:
Crash in only in kernel mode or others?

Reply:
Every mode, it's a bug that currently is masked by what kernel side exit() does